### DISCUSSION TITLE
[Feature] Include subfolders in folder view (ie. view photos within all subfolders)

### DISCUSSION THREAD

**Discussion by alexhk** [immich-app/immich#14426] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

If I am not wrong, currently we can only show photos within a single folder. In the example below it would be great if we click on a year (ie. 2020) and see all photos from the monthly sub-folders (01, 02, 03, etc) for the whole year. 

I hope this is something that can be implemented. Or in case it already is and I am too dumb to figure it out, please let me know!

```/external_libs/photos/
├── 2020/
│   ├── 01
│   ├── 02
│   ├── etc
├── 2021/
│   ├── 01
│   ├── 02
│   ├── etc
└── 2022/...
```

For instance, it could show all photos of the clicked folder and its subfolders below the folder list. When you drill down, into subfolders, of course only these will be shown.

![image](https://github.com/user-attachments/assets/8d652af4-5c9a-4968-879b-9561cc6b7b41)

I started using Immich just recently, not realizing that folder view has just been implemented. Man am I glad we have this feature, without it I'd probably not stay and would have not bought a server license to show my gratitude to the developers.

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by dtczest** [comment]
Thanks for marking mine as a duplicate of this @bo0tzz . I think https://github.com/immich-app/immich/discussions/6519 could also be used to solve this fairly easily.

**Reply by ganymed** [reply]
This was a feature in Apple Aperture and is in Adobe Lightroom Classic, so lots of folks including me have organized their libraries around the ability to the "Toggle show photos of subfolders" feature.  +1 for this feature!!!

**Comment by cb12tre** [comment]
This would be an useful feature - being able to view either a single folder or all subfolders by toggling a selector in the GUI based on user preferences.

**Reply by ManiacDC** [reply]
The way Nextcloud Memories does it, which I really like, it gives an option to open a Timeline view of that folder, which includes all subfolders. The timeline view is a normal timeline view, but filtered down to that specific folder.

**Reply by cb12tre** [reply]
I’ve never used Nextcloud Memories, but I think it would be very convenient if implemented this way.
Hopefully the developers will consider it for a future release.
